### ISSUE TITLE
Custom grains aren't available to pillars

### ISSUE THREAD

**Issue by Dravu** [saltstack/salt#26070]
### Problem Description
I'm using salt-ssh and have a roster file like so:
```
WEB-18_20:
  host: 10.76.18.20
  user: root
  passwd: password
  grains:
    roles:
      - webserver
    location: '2.14'
```
Despite having the custom grain *roles* listed in the roster file, it's not available to pillar files to use. When I turn on debug logging, it appears that only the built-in grains are being provided.
### Examples
While I can see custom grains if I use a command such as:
```
[root@localhost tmp]# salt-ssh -i '*' grains.get roles
WEB-18_20:
    - webserver
```
This isn't the case if I try to retrieve custom grains in pillars. In pillars, I only have access to the built-in grains provided by Salt. If I try to do either of the following, they fail:
top.sls
```
base:
  'G@roles:webserver':
    - match: grain
    - release
```
release.sls
```
release-rpm:
{% if 'webserver' in salt['grains.get']('roles') %}
  - latest: "salt://rpms/webserver-release-1.0-1.noarch.rpm"
{% endif %}
```

### Version Information
```
[root@englab_bootserver tmp]# salt --versions-report
Salt Version:
           Salt: 2015.8.0rc3

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.3.1
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
           RAET: Not Installed
        Tornado: 4.1
            ZMQ: 3.2.5
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
       pycrypto: 2.6.1
        timelib: Not Installed

System Versions:
           dist: centos 7.1.1503 Core
        machine: x86_64
        release: 3.10.0-229.1.2.el7.x86_64
         system: CentOS Linux 7.1.1503 Core
```

**Comment by jfindlay** [comment]
@Dravu, thanks for the report.

**Comment by twiebe** [comment]
This seems to affect other kinds of targeting as well:

    $ sudo salt-ssh alfred grains.get roles
    alfred:
        - webserver
    $ sudo salt-ssh -G roles:webserver test.ping
    No hosts found with target roles:webserver of type grain

**Comment by jgmchan** [comment]
@twiebe I'm pretty sure grains targeting doesn't work for salt-ssh.

https://docs.saltstack.com/en/stage/topics/ssh/
```
Due to the fact that the targeting approach differs in salt-ssh, only glob and 
regex targets are supported as of this writing, the remaining target systems 
still need to be implemented.
```

I am experiencing the same issue with grains and pillars though, which I presume should actually work. I'm using salt-ssh version `2015.5.2`.

**Comment by slav0nic** [comment]
custom grains also not available in jinja {{ grains }}, but available in states

**Comment by slav0nic** [comment]
problem in minion file
minion file after `test.ping`:
```
root@gs1:/tmp/.root_659579_salt# cat minion 

grains:
  game: csgo
  instances: 3
  roles: [gameserv]
id: gs1
log_file: salt-call.log
root_dir: /tmp/.root_659579_salt/running_data
sock_dir: 
```
and after `highstate`:
```
root@gs1:/tmp/.root_659579_salt# cat minion 

{id: gs1, log_file: salt-call.log, root_dir: /tmp/.root_659579_salt/running_data,
  sock_dir: /}
```
maybe problem in https://github.com/saltstack/salt/blob/832870264a76da023c442c8c8f1d2f6e932ff4d6/salt/client/ssh/__init__.py#L789

**Comment by slav0nic** [comment]
fixed :dancers: , but required core devs review
```diff
diff --git a/salt/client/ssh/wrapper/state.py b/salt/client/ssh/wrapper/state.py
index b1135de..98c7d04 100644
--- a/salt/client/ssh/wrapper/state.py
+++ b/salt/client/ssh/wrapper/state.py
@@ -109,6 +109,7 @@ def sls(mods, saltenv='base', test=None, exclude=None, env=None, **kwargs):
             __opts__,
             cmd,
             fsclient=__context__['fileclient'],
+            minion_opts=__salt__.minion_opts,
             **st_kwargs)
     single.shell.send(
             trans_tar,
@@ -175,6 +176,7 @@ def low(data, **kwargs):
             __opts__,
             cmd,
             fsclient=__context__['fileclient'],
+            minion_opts=__salt__.minion_opts,
             **st_kwargs)
     single.shell.send(
             trans_tar,
@@ -239,6 +241,7 @@ def high(data, **kwargs):
             __opts__,
             cmd,
             fsclient=__context__['fileclient'],
+            minion_opts=__salt__.minion_opts,
             **st_kwargs)
     single.shell.send(
             trans_tar,
@@ -332,6 +335,7 @@ def highstate(test=None, **kwargs):
             __opts__,
             cmd,
             fsclient=__context__['fileclient'],
+            minion_opts=__salt__.minion_opts,
             **st_kwargs)
     single.shell.send(
             trans_tar,
@@ -403,6 +407,7 @@ def top(topfn, test=None, **kwargs):
             __opts__,
             cmd,
             fsclient=__context__['fileclient'],
+            minion_opts=__salt__.minion_opts,
             **st_kwargs)
     single.shell.send(
             trans_tar,
@@ -627,6 +632,7 @@ def single(fun, name, test=None, **kwargs):
             __opts__,
             cmd,
             fsclient=__context__['fileclient'],
+            minion_opts=__salt__.minion_opts,
             **st_kwargs)
 
     # Copy the tar down
```

**Comment by jfindlay** [comment]
@basepi

**Comment by basepi** [comment]
@slav0nic I think your proposed patch should be great! Would you mind submitting a pull request?

**Comment by DanielDent** [comment]
I'm having the same issue with salt-ssh grains not showing up in Jinja templates. It would be great to see this patch get merged :)

**Comment by slav0nic** [comment]
@DanielDent you can manually apply patch and check is problem gone

**Comment by DanielDent** [comment]
@slav0nic I have applied your patch, it's working for me. Thank you :)

**Comment by basepi** [comment]
This was fixed and will be in 2016.3.0
